### ISSUE TITLE
Overflows & crashes in OS X CI

### ISSUE THREAD

**Issue by rmcgibbo** [mdtraj/mdtraj#1572]
It appears that on June 14, the test suite ran  fine, with the OS X build+test taking 13 minutes, whereas in the next invokation on July 23, the OS X test suite started taking 60+ minutes and timing out (while the Linux test times were unchanged).

https://dev.azure.com/rmcgibbo/mdtraj/_build?definitionId=1&_a=summary

**Comment by rmcgibbo** [comment]
From: https://dev.azure.com/rmcgibbo/mdtraj/_build/results?buildId=251&view=logs&j=0a2947bc-4ed7-5966-e5d8-17037b1704c1&t=87f4673e-a130-525e-5b37-f43ee4811e8d
```
2020-08-03T21:01:10.6425610Z tests/test_geometry.py::test_angle_triclinic_0 PASSED                    [ 16%]
2020-08-03T21:01:10.6426030Z tests/test_geometry.py::test_angle_1 PASSED                              [ 16%]
2020-08-03T21:53:30.6992280Z ##[error]The operation was canceled.
2020-08-03T21:53:30.7017450Z ##[section]Finishing: Build and test
```

~8 minutes into the build, at 21:01, it looks like it ran thetest_geometry.py tests and then proceeded to simply hang for the remaining 52 minutes in the build.

**Comment by rmcgibbo** [comment]
```
2020-08-04T17:22:21.2143820Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/
python3.7/site-packages/mdtraj/core/trajectory.py:433: in load
2020-08-04T17:22:21.2144550Z     value = loader(filename, **kwargs)
2020-08-04T17:22:21.2144880Z mdtraj/formats/xtc/xtc.pyx:168: in mdtraj.formats.xtc.load_xtc
2020-08-04T17:22:21.2145160Z     ???
2020-08-04T17:22:21.2145560Z mdtraj/formats/xtc/xtc.pyx:175: in mdtraj.formats.xtc.load_xtc
2020-08-04T17:22:21.2145820Z     ???
2020-08-04T17:22:21.2146110Z mdtraj/formats/xtc/xtc.pyx:339: in mdtraj.formats.xtc.XTCTrajectoryFile.read_as_traj
2020-08-04T17:22:21.2146410Z     ???
2020-08-04T17:22:21.2146690Z mdtraj/formats/xtc/xtc.pyx:408: in mdtraj.formats.xtc.XTCTrajectoryFile.read
2020-08-04T17:22:21.2146980Z     ???
2020-08-04T17:22:21.2147400Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2020-08-04T17:22:21.2147650Z 
2020-08-04T17:22:21.2147810Z >   ???
2020-08-04T17:22:21.2148160Z E   MemoryError: Unable to allocate 95.4 TiB for an array with shape (3492404224, 2504, 3) and data type float32
```
👀

**Comment by rmcgibbo** [comment]
The log here is 49MB of integer overflow + logging garbage as well: https://dev.azure.com/rmcgibbo/mdtraj/_build/results?buildId=256&view=logs&s=859b8d9a-8fd6-5a5c-6f5e-f84f1990894e&j=0a2947bc-4ed7-5966-e5d8-17037b1704c1

```
2020-08-04T17:22:21.2211660Z ----------------------------- Captured stdout call -----------------------------
2020-08-04T17:22:21.2211970Z xtc
2020-08-04T17:22:21.2212080Z 
2020-08-04T17:22:21.2212360Z converted 6 frames, 5 atoms
2020-08-04T17:22:21.2212600Z converted 12 frames, 5 atoms
2020-08-04T17:22:21.2212910Z converted 18 frames, 5 atoms
2020-08-04T17:22:21.2213220Z converted 24 frames, 5 atoms
2020-08-04T17:22:21.2213460Z converted 30 frames, 5 atoms
[...]
2020-08-04T17:22:44.4766200Z converted 4799736 frames, 5 atoms
2020-08-04T17:22:44.4766460Z converted 4799742 frames, 5 atoms
2020-08-04T17:22:44.4766710Z converted 4799748 frames, 5 atoms
```

🤣

**Comment by rmcgibbo** [comment]
Other highlights from the 49MB log file:
```
2020-08-04T17:22:44.5755630Z E   MemoryError: Unable to allocate 8.57 TiB for an array with shape (2354937069568,) and data type int32
2020-08-04T17:22:44.5765100Z E   MemoryError: Unable to allocate 90.5 TiB for an array with shape (376789958656, 22, 3) and data type float32
2020-08-04T17:22:44.5771600Z E   MemoryError: Unable to allocate 38.2 TiB for an array with shape (158930305024, 22, 3) and data type float32
2020-08-04T17:22:44.5778500Z E   MemoryError: Unable to allocate 90.5 TiB for an array with shape (376789958656, 22, 3) and data type float32
2020-08-04T17:22:44.5818290Z E   MemoryError: Unable to allocate 90.5 TiB for an array with shape (376789958656, 22, 3) and data type float32
2020-08-04T17:22:44.5893030Z E   MemoryError: Unable to allocate 30.8 TiB for an array with shape (941974880256, 3, 3) and data type float32
2020-08-04T17:22:44.5900470Z E   MemoryError: Unable to allocate 30.8 TiB for an array with shape (941974880256, 3, 3) and data type float32
2020-08-04T17:22:44.5906940Z E   MemoryError: Unable to allocate 113. TiB for an array with shape (941974880256, 11, 3) and data type float32
2020-08-04T17:22:44.5913300Z E   MemoryError: Unable to allocate 103. TiB for an array with shape (429496729600, 22, 3) and data type float32
2020-08-04T17:22:44.5949170Z E   MemoryError: Unable to allocate 90.5 TiB for an array with shape (376789958656, 22, 3) and data type float32
2020-08-04T17:22:44.5958370Z E   MemoryError: Unable to allocate 86.1 TiB for an array with shape (788527316992, 10, 3) and data type float32
2020-08-04T17:22:44.5967560Z E   MemoryError: Unable to allocate 86.1 TiB for an array with shape (788527316992, 10, 3) and data type float32
2020-08-04T17:22:44.5976310Z E   MemoryError: Unable to allocate 61.0 TiB for an array with shape (1864466169856, 3, 3) and data type float32
2020-08-04T17:22:44.5981380Z E   MemoryError: Unable to allocate 90.5 TiB for an array with shape (376789958656, 22, 3) and data type float32
2020-08-04T17:22:44.5988160Z E   MemoryError: Unable to allocate 90.5 TiB for an array with shape (376789958656, 22, 3) and data type float32
2020-08-04T17:22:44.5997090Z E   MemoryError: Unable to allocate 89.3 TiB for an array with shape (81852071936, 100, 3) and data type float32
```

**Comment by mattwthompson** [comment]
There goes the  dream of sneaking jobs requiring 100 TB of memory through Azure unnoticed ...

**Comment by rmcgibbo** [comment]
The apparent commit that triggered this was a change in the docs https://github.com/mdtraj/mdtraj/commit/86294e9a72ce197b8202109e2e67978e55ea3ec3. I assume that there must have been a change in the compiler versions or something(?!) that exposed some kind of integer overflow bug?

**Comment by rmcgibbo** [comment]
There are a variety of differences between minor library versions and stuff since they ran over a month apart, but I assume that clang9 -> clang10 is the relevant change.

Before: https://dev.azure.com/rmcgibbo/mdtraj/_build/results?buildId=247&view=logs&j=0a2947bc-4ed7-5966-e5d8-17037b1704c1&t=87f4673e-a130-525e-5b37-f43ee4811e8d
```## Package Plan ##

  environment location: /usr/local/miniconda/conda-bld/mdtraj_1592172925478/_build_env


The following NEW packages will be INSTALLED:

    cctools:            927.0.2-h5ba7a2e_4       conda-forge
    clang:              9.0.1-default_hf57f61e_0 conda-forge
    clang-tools:        9.0.1-default_hf57f61e_0 conda-forge
    clang_osx-64:       9.0.1-h05bbb7f_2         conda-forge
    clangxx:            9.0.1-default_hf57f61e_0 conda-forge
    clangxx_osx-64:     9.0.1-h05bbb7f_2         conda-forge
    compiler-rt:        9.0.1-h6a512c6_3         conda-forge
    compiler-rt_osx-64: 9.0.1-h99342c6_3         conda-forge
    ld64:               450.3-h3c32e8a_4         conda-forge
    libcxx:             10.0.0-h1af66ff_2        conda-forge
    libllvm9:           9.0.1-h7475705_1         conda-forge
    llvm-tools:         9.0.1-h1341992_1         conda-forge
    llvmdev:            9.0.1-h7475705_1         conda-forge
    tapi:               1000.10.8-ha1b3eb9_4     conda-forge
    zlib:               1.2.11-h0b31af3_1006     conda-forge
```

After: https://dev.azure.com/rmcgibbo/mdtraj/_build/results?buildId=248&view=logs&j=0a2947bc-4ed7-5966-e5d8-17037b1704c1&t=87f4673e-a130-525e-5b37-f43ee4811e8d

```## Package Plan ##

  environment location: /usr/local/miniconda/conda-bld/mdtraj_1595527599773/_build_env


The following NEW packages will be INSTALLED:

    cctools_osx-64:     949.0.1-h5ba7a2e_1        conda-forge
    clang:              10.0.1-default_hf57f61e_0 conda-forge
    clang-tools:        10.0.1-default_hf57f61e_0 conda-forge
    clang_osx-64:       10.0.1-h05bbb7f_0         conda-forge
    clangxx:            10.0.1-default_hf57f61e_0 conda-forge
    clangxx_osx-64:     10.0.1-h05bbb7f_0         conda-forge
    compiler-rt:        10.0.1-he6db49b_0         conda-forge
    compiler-rt_osx-64: 10.0.1-h033240e_0         conda-forge
    ld64_osx-64:        530-h3c32e8a_1            conda-forge
    libclang-cpp10:     10.0.1-default_hf57f61e_0 conda-forge
    libcxx:             10.0.1-h5f48129_0         conda-forge
    libllvm10:          10.0.1-h009f743_0         conda-forge
    llvm-tools:         10.0.1-h1341992_0         conda-forge
    llvmdev:            10.0.1-h009f743_0         conda-forge
    tapi:               1000.10.8-ha1b3eb9_4      conda-forge
    zlib:               1.2.11-h0b31af3_1006      conda-forge
```

**Comment by jag1g13** [comment]
I've just had a rather strange set of CI jobs for my code which depends on MDTraj, that might add a bit more information to this.

In a run yesterday I had jobs succeeding on `macos-latest` (except with Python 3.9, but that's a separate issue)
See: https://github.com/jag1g13/pycgtool/runs/1317536345?check_suite_focus=true

Today, I've got failures for all versions of Python with this strange array sizing error you've seen previously. See: https://github.com/jag1g13/pycgtool/runs/1322811333?check_suite_focus=true

The only change to my code during this time was to disable fail-fast. The log during the setup phase indicates that these two runs used different environment configurations: 20201017.1 worked, 20201026.2 did not. The difference between these two configuration is in this commit: https://github.com/actions/virtual-environments/pull/1922/commits/c09dca28df69d9aaaeac5635257d23722810d307, which does show an update to clang. Strangely, it's clang 10.0.1 that worked and 11.0.0 that didn't.

I'm happy to assist with debugging this if there's anything you'd like me to test.

**Comment by rmcgibbo** [comment]
Unfortunately, I haven't really had any time to look into this, and I don't particularly know how to. I don't have access to an OS X machine with more of the more recent OSes & clang versions. I think it's probably necessary to reproduce this behavior on a workstation rather than a CI instance so that it can be interrogated with gdb/lldb and the like.

**Comment by rmcgibbo** [comment]
This should be fixed on master!
